src/executor/evaluate/mod.rs

```diff
@@ -320,6 +320,22 @@ async fn evaluate_function<'a, T: 'static + Debug>(
             Nullable::Null => Ok(Value::Null),
         }
         .map(Evaluated::from),
+        Function::Sin(expr) | Function::Cos(expr) | Function::Tan(expr) => {
+            let float_number = eval_to_float(func.name(), expr).await?;
```

Comment:
I tried to adding display to Function, but i'm failed due to a lifetime problem.
The return value of to_string() is String, but the eval_to_float functions parameter is &static(now i'm changed to 'a) str, so temporary value is freed in the statement.
I tried a lot but failed, how can i solve this problem?

Reply:
What you've done is actually more meaningful :)
Then, you don't even need to pass `func.name()` at all.

1. Applying `strum` crate
You can use https://crates.io/crates/strum_macros
```rust
use {
    ..
    strum::Display,                                                          
};                                                                           
                                                                             
#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, Display)]
pub enum Function {                                                          
    #[strum(to_string = "LOWER")]                                            
    Lower(Expr),                                                             
                                                                             
    #[strum(to_string = "UPPER")]                                            
    Upper(Expr),                                                             
                                                                             
    #[strum(to_string = "LEFT")]                                             
    Left { expr: Expr, size: Expr },                                         
                                                                             
    #[strum(to_string = "RIGHT")]                                            
    Right { expr: Expr, size: Expr },                                        
                                                                             
    #[strum(to_string = "CEIL")]                                             
    Ceil(Expr),                                                              
                                                                             
    #[strum(to_string = "ROUND")]                                            
    Round(Expr),                 

    ..
}                                      
```

Then, `func.to_string()` will work.

2. Better `eval_to_float`
Thanks to `Display`, now `eval_to_float` does not require function name.

```rust
let eval_to_float = |expr| async move {                                                          
    match eval(expr).await?.try_into()? {                                                        
        Value::I64(v) => Ok(Nullable::Value(v as f64)),                                          
        Value::F64(v) => Ok(Nullable::Value(v)),                                                 
        Value::Null => Ok(Nullable::Null),                                                       
        _ => Err::<_, Error>(EvaluateError::FunctionRequiresFloatValue(func.to_string()).into()),
    }                                                                                            
};                                                                                               
```

Reply:
Thanks for your helps! It was a simple solution by removing unnecessary parameters.
I also removed the parameter of other eval functions in the same way.